framework/PostProcessors/ParetoFrontierPostProcessor.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on March 25, 2020
+
+@author: mandd
+"""
+from __future__ import division, print_function , unicode_literals, absolute_import
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import utils
+from utils import InputData, InputTypes
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+class ParetoFrontier(PostProcessor):
+  """
+    This postprocessor selects the points that lie on the Pareto frontier
+    The postprocessor acts only on PointSet and return a subset of such PointSet
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ParetoFrontier, cls).getInputSpecification()
+    inputSpecification.addSub(InputData.parameterInputFactory('costID' , contentType=InputTypes.StringType))
+    inputSpecification.addSub(InputData.parameterInputFactory('valueID', contentType=InputTypes.StringType))
+    return inputSpecification
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the PostProcessor
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
```

Comment:
remove the def initialize since you are not "extending" it... (it will automatically takes the PostProcessor one)